TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -474,5 +474,154 @@ private SymbolReference CreateTableOrAliasName(CodeElementsParser.TableOrAliasNa
+            {
+                diagnosticIdLiteral = new SqlConstant(ParseTreeUtils.GetTokenFromTerminalNode(context.IntegerLiteral()));
+            }
+            foreach (var assignment in context.repeatedConnectionOrConditionInformation())
+            {
+                assignments.Add(CreateSingleInformationAssignment(assignment));
+            }
+            return new ConditionInformation(diagnosticIdVariable, diagnosticIdLiteral, assignments);
+        }
+
+        private SingleInformationAssignment CreateSingleInformationAssignment(CodeElementsParser.RepeatedConnectionOrConditionInformationContext context)
+        {
+            SymbolReference itemName = null;
+            SqlVariable storage = null;
+            if (context.variable_3 != null)
+            {
+                storage = CreateSqlVariable(context.variable_3);
+            }
+
+            if (context.UserDefinedWord() != null)
+            {
+                itemName = CreateSymbolReference(ParseTreeUtils.GetTokenFromTerminalNode(context
+                    .UserDefinedWord()));
+            }
+
+            return new SingleInformationAssignment(itemName, storage);
+        }
+
+        private CombinedInformation CreateCombinedInformationClause(CodeElementsParser.CombinedInformationClauseContext context)
+        {
+            SqlVariable storage = null;
+            var items = new List<CombinedInformationItem>();
+            if (context.variable_4 != null)
+            {
+                storage = CreateSqlVariable(context.variable_4);
+            }
+            foreach (var combinedInformationItem in context.repeatedCombinedInformation())
+            {
+                items.Add(CreateCombinedInformationItem(combinedInformationItem));
+            }
+            return new CombinedInformation(storage, items);
+        }
+
+        private CombinedInformationItem CreateCombinedInformationItem(
+            CodeElementsParser.RepeatedCombinedInformationContext context)
+        {
+            if (context.SQL_STATEMENT() != null)
+            {
+                return new CombinedInformationItem(CombinedInformationItemType.Statement, null, null);
+            }
+
+            else if (context.SQL_CONDITION() != null)
+            {
+                CreateConnectionOrConditionInformation(out var diagnosticIdVariable, out var diagnosticIdLiteral);
+                return new CombinedInformationItem(CombinedInformationItemType.Condition, diagnosticIdVariable, diagnosticIdLiteral);
+            }
+            else if (context.SQL_CONNECTION() != null)
+            {
+                CreateConnectionOrConditionInformation(out var diagnosticIdVariable, out var diagnosticIdLiteral);
+                return new CombinedInformationItem(CombinedInformationItemType.Connection, diagnosticIdVariable, diagnosticIdLiteral);
+            }
+
+            void CreateConnectionOrConditionInformation(out SqlVariable diagnosticIdVariable,
```

Comment:
- naming is incorrect as this local function does not create any information item but a variable or a constant. Please rename to `CreateDiagnosticId`
- the use of out variables is redundant. As it is a local function, you can declare `diagnosticIdVariable` and `diagnosticIdLiteral` in parent method and `CreateDiagnosticId` will be able to write them directly so no need for any parameter